### DISCUSSION TITLE
Adding vector search feature available in manticore

### DISCUSSION THREAD

**Discussion by ilyasch199** [jsuto/piler#172] [Q&A]
Hello Mail Piler community,

I'd like to propose enhancing Mail Piler's search functionality by integrating the new vector search capabilities of ManticoreSearch (added in May 2024). This integration could significantly improve search relevance and enable more advanced semantic search features.

i would also like to know which part of the code is responsible for indexing emails into manticore database so i can try to add this feature.

**Comment by jsuto** [comment]
Thank you for the idea. src/tokenizer.c writes a few buffers with the data to be indexed. Regarding the message body (including the attachments as well) the tokens (or words) are written to the buffer at https://github.com/jsuto/piler/blob/master/src/tokenizer.c#L136

**Comment by jsuto** [comment]
I wonder if there's a progress with the vector search stuff?

**Comment by jsuto** [comment]
Not much update on this so far. Closing this discussion.
